### PR TITLE
core/metrics.py: add the metrics framework

### LINKED ISSUE #83: Design and create metrics framework to export the service metrics

### PR DESCRIPTION
**Pull request by leyao-daily**
This PR is to fix #83 , it will create a common metrics framework to export the service metrics.

### CODE REVIEW

File: cnap/core/metrics.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    This enum defines the error messages that can be raised.
+    """
+    METRIC_EXISTS = 'Metric {} already exists'
+    INVALID_METRIC_TYPE = 'Invalid metric type: {}'
+    METRIC_DOES_NOT_EXIST = 'Metric {} does not exist'
+    METRIC_TYPE_MISMATCH = 'Metric {} is not a {}'
+
+
+class MetricsManager:
```
Comment (Hulongyin) [inline cnap/core/metrics.py:40]: Does this class need to use the singleton pattern to ensure that there is only one instance in each microservice?

File: cnap/core/metrics.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+        Raises:
+            ValueError: If the specified metric does not exist or the metric type is invalid.
+        """
+        with self.lock:
+            metric = self.metrics.get(metric_name)
+            if metric is None:
+                raise ValueError(ErrorMessage.METRIC_DOES_NOT_EXIST.value.format(metric_name))
+
+            if not isinstance(metric, metric_type):
```
Comment (Hulongyin) [inline cnap/core/metrics.py]: `metric_type` doesn't seem to be a suitable argument for the `isinstance` function.
TypeError: isinstance() arg 2 must be a type or tuple of types

File: cnap/core/metrics.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            metric_name (str): The name of the Gauge metric to set.
+            value (float): The value to set the Gauge to.
+
+        Raises:
+            ValueError: If the specified metric does not exist or is not a Gauge.
+        """
+        metric = self.get_metric(metric_name, Gauge)
+        metric.set(value)
+
+    def observe_summary(self, metric_name: str, value: float) -> None:
```
Comment (Hulongyin) [inline cnap/core/metrics.py:183]: Do we need a `observe_histogram` function for `Histogram` type?

File: cnap/core/metrics.py (hunk 1 above)
Comment (leyao-daily) [reply cnap/core/metrics.py:40]: It makes sense

File: cnap/core/metrics.py (hunk 2 above)
Comment (leyao-daily) [reply cnap/core/metrics.py]: fixed.

**Review by Hulongyin — APPROVED** [review]
LGTM. Thanks

**Pull request merged**